tests/testthat/test-target_market_share.R

```diff
@@ -1143,3 +1143,24 @@ test_that("`target_market_share` outputs only positive values of `production`(#3
 
   expect_false(any(out$production < 0))
 })
+
+test_that("NA in technology share (ADO876)", {
+  ald <- tibble::tribble(
+    ~company_id, ~name_company, ~is_ultimate_owner,  ~sector,           ~technology, ~year, ~plant_location, ~production,   ~production_unit,  ~emission_factor,               ~emission_factor_unit, ~ald_timestamp,
+    38998,  "parc eolien de bel air",              FALSE, "power", "renewablescap",  2020,            "FR",      0, "MW",  0.74613363337134, "tonnes of CO2 per MWh",      "2021 Q1",
+    38998,  "parc eolien de bel air",              FALSE, "power", "renewablescap",  2021,            "FR",      1, "MW",  0.74613363337134, "tonnes of CO2 per MWh",      "2021 Q1"
+  )
+  matched <- tibble::tribble(
+    ~id_loan, ~loan_size_outstanding, ~loan_size_outstanding_currency, ~loan_size_credit_limit, ~loan_size_credit_limit_currency, ~id_2dii,            ~level, ~score,      ~sector,      ~name_ald,  ~sector_ald,
+    "L162",                      1,                           "EUR",                       2,                            "EUR",    "UP1", "ultimate_parent",      1, "power", "parc eolien de bel air", "power"
+  )
+  out <- target_market_share(
+    matched,
+    ald,
+    scenario_demo_2020,
+    region_isos_demo)
+
+  expect_equal(any(out$technology_share),"NA")
```

Comment:
This isn't testing for what you think it is testing for. 
What this test is saying is: 
`any(out$technology_share)` asks if any of the values of `technology_share` are `TRUE` (which doesn't make sense, since `technology_share` is numeric.). So this condition will always output `FALSE`

Then `expect_equal(FALSE, "NA")` will always fail, since `FALSE != "NA"`. 

Try to think of a more specific test. This is always the MOST crucial part of the debugging process (IMO), so it's important to think carefully about this at the beginning.

The process should be:

* add failing test
* debug
* submit PR once this test passes

If you look at the CHECKS on this PR, you will see that they all fail, since this PR is failing.

Reply:
Before approving, this test will need to be something that: 
* fails on `master` version of `r2dii.analysis`
* passes on this branches version of `r2dii.analysis`